services/api/src/resources/problem/sql.ts

```diff
@@ -128,14 +128,19 @@ export const Sql = {
         created
       })
       .toString(),
-  deleteProblem: (environment, identifier) =>
-    knex('environment_problem')
+  deleteProblem: (environment, identifier, service) => {
+    let q = knex('environment_problem')
       .where({
         environment: environment,
         identifier: identifier
-      })
-      .del()
-      .toString(),
+      });
+
+    if (service != undefined) {
+      q.where('lagoon_service', service);
+    }
+
+    return q.del().toString()
+  },
```

Comment:
Minor formatting nit - :see_no_evil: at the rest of the file(s)

```suggestion
      });
    if (service != undefined) {
      q.where('lagoon_service', service);
    }
    return q.del().toString();
  },
```